pkg/apic/provisioning/requeststatus.go

```diff
@@ -0,0 +1,69 @@
+package provisioning
+
+import "fmt"
+
+const statusSetError = "can not set status as it's already been set"
+
+// RequestStatus - holds info about the status of the request
+type RequestStatus struct {
+	status  status
+	message string
+}
+
+// RequestStatusBuilder - builder to create new request status
+type RequestStatusBuilder interface {
+	Success() RequestStatusBuilder
+	Failed(message string) RequestStatusBuilder
+	Process() (*RequestStatus, error)
```

Comment:
Returning a pointer here however the provisioning interface are returning an object

Reply:
Removed this